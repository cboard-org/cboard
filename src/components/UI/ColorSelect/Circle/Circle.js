import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  /**
   * Stroke width
   */
  strokeWidth: PropTypes.number,
  /**
   * Fill color
   */
  fill: PropTypes.string
};

const defaultProps = {
  fill: 'transparent',
  strokeWidth: 0
};

function Circle(props) {
  const { fill, strokeWidth } = props;

  return (
    <svg height="48" width="48">
      <circle
        cx="24"
        cy="24"
        r="15"
        fill={fill}
        stroke="black"
        strokeWidth={strokeWidth}
      />
    </svg>
  );
}

Circle.propTypes = propTypes;
Circle.defaultProps = defaultProps;

export default Circle;
