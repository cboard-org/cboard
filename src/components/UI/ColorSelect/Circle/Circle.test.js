import React from 'react';
import { shallowMatchSnapshot } from '../../../../common/test_utils';

import Circle from './Circle';

describe('Circle tests', () => {
  test('default renderer', () => {
    shallowMatchSnapshot(<Circle />);
  });
});
