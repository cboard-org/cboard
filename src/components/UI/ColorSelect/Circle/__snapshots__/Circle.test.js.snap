// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle tests default renderer 1`] = `
<svg
  height="48"
  width="48"
>
  <circle
    cx="24"
    cy="24"
    fill="transparent"
    r="15"
    stroke="black"
    strokeWidth={0}
  />
</svg>
`;
